src/Tmds.Ssh/SocketSshConnection.cs

```diff
@@ -1,250 +1,27 @@
-﻿// This file is part of Tmds.Ssh which is released under MIT.
+// This file is part of Tmds.Ssh which is released under MIT.
 // See file LICENSE for full license details.
 
-using System.Buffers;
-using System.Diagnostics;
 using System.Net.Sockets;
-using System.Text;
 using Microsoft.Extensions.Logging;
 
 namespace Tmds.Ssh;
 
-sealed class SocketSshConnection : SshConnection
+sealed class SocketSshConnection : StreamSshConnection
```

Comment:
Can you remove this class completely? The NetworkStream that wraps the Socket can own it (for dispose).

Reply:
I'm not sure. The kept it as a sub class because I assumed we needed to keep the `Socket` object referenced somewhere. The socket stream is technically a separate object and I assumed that once the GC cleans up the Socket object it will close the stream as well but if that's not the case I can certainly simplify things.

Reply:
`NetworkStream` will take ownership of the `Socket`, which means it will dispose the Socket when it gets disposed.

```
NetworkStream (System.Net.Sockets.Socket socket, bool ownsSocket);
```

As long as `StreamSshConnection` is considered reachable, the `Socket` is considered reachable too.

More technical:

The GC scans all the memory and follows locations it knows to be managed references. Anything not reachable from a stack is considered collectible. Objects that have a finalizer (like `~Socket` and `~SafeSocketHandle`) get finalized before being collected. If the user had called `Dispose` on these objects (for example through a `using StreamSshConnection` -> `NetworkStream.Dispose` -> `Socket.Dispose` -> `SafeSocketHandle.Dispose`) then that `Dispose` should included a call to `GC.SuppressFinalize` (on objects with a finalizer) which tells the GC it no longer needs to run the finalizer before reclaiming the memory on that object.

Reply:
Thanks for the info, I was confusing this with [TcpClient.GetStream](https://learn.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient.getstream?view=net-9.0) where I was unsure who was the owner of the the actual underlying socket and what needed to be kept in referenced. I've changed it to rely on the `NetworkStream` ownership as you've mentioned here which tidies up the code quite nicely.